pkg/v1/tkg/client/validate_upgrade.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"fmt"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/constants"
+)
+
+var (
+	azureVarsToBeValidated = []string{constants.ConfigVariableAzureClientSecret}
+)
+
+// ValidateEnvVariables validates the presence of required environment variables for a given IaaS
+func (c *TkgClient) ValidateEnvVariables(iaas string) error {
+	// This function only contains a validator for Azure environment variables for now
+	// This function can be used to add validators for environment variables pertaining to other IaaSes in the future
+	// as and when required
+	if iaas == AzureProviderName {
+		return c.validateAzureEnvVariables(azureVarsToBeValidated)
+	}
+	return nil
+}
+
+func (c *TkgClient) validateAzureEnvVariables(azureEnvVars []string) error {
+	for _, envVar := range azureEnvVars {
+		_, err := c.TKGConfigReaderWriter().Get(envVar)
+		if err != nil {
+			return fmt.Errorf("config Variable %s not set", envVar)
```

Comment:
Tests don't cover the case when an error occurs. 

When AZURE_CLIENT_SECRET is not set, we _want_ it to fail, right? If we do, we need to add this as a test case.

Reply:
I added a unit test case for this - https://github.com/vmware-tanzu/tanzu-framework/pull/1841/files#diff-1c1b46de79e151a439d38ef7a2d3e0c85de6747348af329010dc5acd74eade92R68